autopilot/contractor.go

```diff
@@ -575,16 +586,29 @@ func (c *contractor) runContractChecks(ctx context.Context, w Worker, contracts
 
 		// if the host doesn't have a valid pricetable, update it
 		var invalidPT bool
-		if err := refreshPriceTable(ctx, w, &host.Host); err != nil {
+		stalled, err := refreshPriceTable(ctx, w, &host.Host)
+		if err != nil {
 			c.logger.Errorf("could not fetch price table for host %v: %v", host.PublicKey, err)
 			invalidPT = true
 		}
 
+		// if we were stalled refreshing the price table, we fetch the consensus
+		// state and recreate the gouging checker to ensure we're checking host
+		// height against an updated renter height
+		if stalled {
```

Comment:
I wouldn't have `refreshPriceTable` return a `stalled` bool. It introduces an unnecessary named return var and doesn't accurately reflect whether we actually got stalled for a while.

Instead we can just fetch the consensus state here for every contract and create a custom gouging checker for each one. Then you also don't need to pass in `cs` at all. 

It won't get any more accurate than that.

Reply:
ok fair enough, I was happy to pass `cs` through method params but I can see why that's not worth it